ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/AbstractDebugTraceBlock.java

```diff
@@ -102,7 +102,7 @@ protected Collection<DebugTraceTransactionResult> getTraces(
             block ->
                 Tracer.processTracing(
                     getBlockchainQueries(),
-                    block.getHash(),
+                    Optional.of(block.getHeader()),
```

Comment:
Good catch. 
I remember having issues with unit tests related to this, but I agree, it is better to use header instead of the hash.